src/pkgTypes.ts

```diff
@@ -0,0 +1,94 @@
+import type { RawPkg, StyleType } from './@types/pkg';
+import type { File } from './jsDelivr';
+import { datadog } from './utils/datadog';
+
+export const styleFileExtensions: StyleType[] = ['css', 'less', 'scss'];
+const jsFileExtensions = ['js', 'mjs', 'cjs'];
+
+function createFilePattern(extensions: string[]): RegExp {
+  const extPattern = extensions.join('|');
+  return new RegExp(
+    `^(?:(?!\\/(docs?|documentation|examples?|samples?|demos?|tests?)\\/)(?!\\/[._]).)+\\.(${extPattern})$`
+  );
+}
+
+export function getStyleTypes(
+  pkg: Pick<RawPkg, 'styleTypes'>,
+  filelist: File[]
+): Pick<RawPkg, 'styleTypes'> {
+  const start = Date.now();
+
+  try {
+    const styleTypes = new Set(pkg.styleTypes);
+
+    for (const file of filelist) {
+      if (!createFilePattern(styleFileExtensions).test(file.name)) {
```

Comment:
Why are we constructing a new regexp here for each file?

Reply:
my bad, I'll move it outside the loop

Reply:
https://github.com/algolia/npm-search/pull/683/commits/7df273f9c2b1292db4920092b3693102efb2a5d4